ext/opcache/jit/zend_jit_trace.c

```diff
@@ -7125,6 +7125,74 @@ static zend_jit_trace_stop zend_jit_compile_root_trace(zend_jit_trace_rec *trace
 	return ret;
 }
 
+/* Set counting handler back to original VM handler. */
+static void zend_jit_shutdown_hot_trace_counters(zend_op_array *op_array)
+{
+	zend_jit_op_array_trace_extension *jit_extension;
+	uint32_t i;
+
+	jit_extension = (zend_jit_op_array_trace_extension*)ZEND_FUNC_INFO(op_array);
+	for (i = 0; i < op_array->last; i++) {
+		/* opline with Jitted code handler is ignored. */
+        if (jit_extension->trace_info[i].trace_flags & \
+		(ZEND_JIT_TRACE_JITED|ZEND_JIT_TRACE_BLACKLISTED)) {
+            continue;
+        }
+		zend_shared_alloc_lock();
+        if (jit_extension->trace_info[i].trace_flags & \
+		(ZEND_JIT_TRACE_START_LOOP | ZEND_JIT_TRACE_START_ENTER | ZEND_JIT_TRACE_START_RETURN)) {
+			SHM_UNPROTECT();
+			op_array->opcodes[i].handler = jit_extension->trace_info[i].orig_handler;
+			SHM_PROTECT();
+        }
+		zend_shared_alloc_unlock();
+    }
+}
+
+/* Get the tracing op_array. */
+static void zend_jit_shutdown_persistent_op_array(zend_op_array *op_array) {
+    zend_func_info *func_info = ZEND_FUNC_INFO(op_array);
+    if (!func_info) {
```

Comment:
Do we really need to check if it is a NULL?

Reply:
> Do we really need to check if it is a NULL?

Though I have not tested this corner case, I follow the original code